include/jemalloc/internal/arena_inlines_b.h

```diff
@@ -60,10 +60,22 @@ arena_prof_info_get(tsd_t *tsd, const void *ptr, alloc_ctx_t *alloc_ctx,
 
 	if (unlikely(!is_slab)) {
 		/* extent must have been initialized at this point. */
-		large_prof_info_get(extent, prof_info);
+		large_alloc_prof_info_get(extent, prof_info);
+
+		if (reset && (uintptr_t)prof_info->alloc_tctx > (uintptr_t)1U) {
+			/*
+			 * This allocation was a prof sample.  Reset the
+			 * pointer on the recent allocation record, so that
+			 * this allocation is recorded as released.
+			 */
+			prof_recent_alloc_reset(tsd, extent);
```

Comment:
Question: should we also reset tctx here? Will that work?

Reply:
Yes, we can, but we don't have to -
- If the allocation will be gone, then there's no more place that's trying to access the `tctx` field before the allocation dies;
- If the allocation is not gone, then the `tctx` field will be properly handled later in `prof_realloc()`.

Reply:
Okay let's reconsider how to refactor this later on. This type of operations should be group together.